src/Data/Path/Pathy.purs

```diff
@@ -505,20 +527,32 @@ module Data.Path.Pathy
   instance eqPath :: Eq (Path a b s) where
     eq p1 p2 = canonicalize p1 `identicalPath` canonicalize p2
 
-  instance showFileName :: Show FileName where
-    show (FileName name) = "FileName " <> show name
+  instance ordPath :: Ord (Path a b s) where
```

Comment:
It's so you can use paths in dictionaries and the like, the ordering isn't meaningful.